app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt

```diff
@@ -86,15 +88,36 @@ sealed interface ReaderScreenArgs : Parcelable {
 @Destination<RootGraph>
 @Composable
 fun ReaderScreen(args: ReaderScreenArgs, navigator: DestinationsNavigator) = composing(navigator) {
-    Deferred({ preparePageLoader(args) }) { pageLoader ->
-        if (pageLoader.isReady) {
-            val info = (args as? ReaderScreenArgs.Gallery)?.info
-            ReaderScreen(pageLoader, info, navigator)
-        } else {
-            val readingFailed = stringResource(R.string.error_reading_failed)
-            LaunchedEffect(Unit) {
-                showSnackbar(readingFailed)
-                popBackStack()
+    Await({ preparePageLoader(args) }) { pageLoader ->
+        DisposableEffect(pageLoader) {
+            pageLoader.start()
+            onDispose {
+                pageLoader.stop()
+            }
+        }
+        val readingFailed = stringResource(R.string.error_reading_failed)
+        Await({
+            kotlin.runCatching {
+                readingFailed.takeUnless { pageLoader.awaitReady() }
+            }.getOrElse {
+                if (it is CancellationException) {
+                    popBackStack()
```

Comment:
We should not propagate cancellation?

Reply:
We should exit reader when the user press back button